server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/controller/CupController.java

```diff
@@ -0,0 +1,68 @@
+package hu.flowacademy.epsilon.sport_event_organizer.controller;
+
+import hu.flowacademy.epsilon.sport_event_organizer.model.Cup;
+import hu.flowacademy.epsilon.sport_event_organizer.model.Team;
+import hu.flowacademy.epsilon.sport_event_organizer.model.User;
+import hu.flowacademy.epsilon.sport_event_organizer.service.CupService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Set;
+
+@RestController
+@RequestMapping(path = "/cup")
+public class CupController {
+
+    @Autowired
+    private CupService cupService;
+
+
+    @PostMapping("/create")
+    public ResponseEntity<Cup> createCup(@RequestBody Cup team) {
+        return ResponseEntity.ok(cupService.save(team));
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Cup> updateCup(@RequestBody Cup cup) {
+        return ResponseEntity.ok(cupService.update(cup));
+    }
+
+    @DeleteMapping("/delete/{cupName}")
+    public ResponseEntity<Void> deleteCup(@PathVariable String cupName) {
+        cupService.deleteCupByName(cupName);
+        return ResponseEntity.noContent().build();
+    }
+
+
+    @GetMapping("/get/{cupName}")
+    public ResponseEntity<Cup> getByCupName(@PathVariable String cupName) {
+        return ResponseEntity.ok(cupService.getByName(cupName));
+    }
+
+    @GetMapping("/get-by-current-organizer")
+    public ResponseEntity<List<Cup>> getCupByOrganizer() {
+        return ResponseEntity.ok(cupService.getByCurrentOrganizer());
+    }
+
+    @PutMapping("/put-organizer/{googleName}/{cupName}")
+    public ResponseEntity<Set<User>> putOrganizer(@PathVariable String googleName, @PathVariable String cupName) {
+        return ResponseEntity.ok(cupService.putOrganizer(googleName, cupName));
```

Comment:
Ha kollekcióval térsz vissza, akkor nem kell ResponseEntity-be wrappelni, elég közvetlenül visszaadni (ilyenkor is 200-as státusszal térünk vissza)